frontend/src/scenes/surveys/surveyActivityDescriber.test.tsx

```diff
@@ -0,0 +1,260 @@
+        const result = describeFieldChange('is active', false, true)
+        expect(getTextContent(result)).toBe('changed is active from false to true')
+    })
+
+    test('handles null values', () => {
+        const result = describeFieldChange('response limit', null, null)
+        expect(getTextContent(result)).toBe('')
+    })
+})
+
+describe('describeCommonChanges', () => {
+    const before: SurveyQuestion = {
+        question: 'What is your favorite color?',
+        description: 'Choose a color',
+        type: SurveyQuestionType.SingleChoice,
+        optional: false,
+        buttonText: 'Next',
+        choices: ['Red', 'Blue', 'Green'],
+    }
+    const after: SurveyQuestion = {
+        ...before,
+        question: 'What is your favorite animal?',
+        description: 'Choose an animal',
+        optional: true,
+        buttonText: 'Continue',
+    }
+
+    test('describes common changes', () => {
+        const changes = describeCommonChanges(before, after)
+        expect(changes).toHaveLength(4)
+        expect(getTextContent(changes[0])).toBe(
+            'changed question text from "What is your favorite color?" to "What is your favorite animal?"'
+        )
+        expect(getTextContent(changes[1])).toBe('updated description')
+        expect(getTextContent(changes[2])).toBe('made question optional')
+        expect(getTextContent(changes[3])).toBe('changed button text from "Next" to "Continue"')
+    })
+})
+
+describe('describeLinkChanges', () => {
+    const before: LinkSurveyQuestion = {
+        question: 'Visit our website',
+        type: SurveyQuestionType.Link,
+        link: 'http://example.com',
+    }
+    const after: LinkSurveyQuestion = {
+        ...before,
+        link: 'http://example.org',
+    }
+
+    test('describes link changes', () => {
+        const changes = describeLinkChanges([before, after])
+        expect(changes).toHaveLength(1)
+        expect(getTextContent(changes[0])).toBe('updated link from http://example.com to http://example.org')
+    })
+})
+
+describe('describeRatingChanges', () => {
+    const before: RatingSurveyQuestion = {
+        question: 'Rate our service',
+        type: SurveyQuestionType.Rating,
+        display: 'emoji',
+        scale: 5,
+        lowerBoundLabel: 'Poor',
+        upperBoundLabel: 'Excellent',
+    }
+    const after: RatingSurveyQuestion = {
+        ...before,
+        display: 'number',
+        scale: 10,
+        lowerBoundLabel: 'Bad',
+        upperBoundLabel: 'Good',
+    }
+
+    test('describes rating changes', () => {
+        const changes = describeRatingChanges([before, after])
+        expect(changes).toHaveLength(3)
+        expect(getTextContent(changes[0])).toBe('changed rating display from emoji to number')
+        expect(getTextContent(changes[1])).toBe('changed rating scale from 5 to 10')
+        expect(getTextContent(changes[2])).toBe('updated rating labels from Poor-Excellent to Bad-Good')
```

Comment:
Multi-word labels will be rendered like this `...from Very bad-Very good to Quite bad-Quite good`

I'd suggest double quotes here: `...from "Very bad"-"Very good" to "Quite bad"-"Quite good"`